src/text2.c

```diff
@@ -3210,3 +3210,860 @@ void sub_80089AC(const UnkTextStruct2 *r4, UnkTextStruct2_sub *r5_Str)
+                if (*str == 0x5D) {
+                    str++;
+                    break;
+                }
+                str++;
+            }
+            if (sp->unk24 != NULL) {
+                const u8 *strNew = sp->unk24(strBefore, str, sp);
+                if (strNew != NULL) {
+                    str = strNew;
+                }
+            }
+
+            if (sp->unk21 != 0)
+                break;
+        }
+        else if (str[1] == 0x3D) {
+            sp->unk0 = str[2];
+            str += 3;
+            if (*str == 0x2E)
+                str++;
+        }
+        else if (str[1] == 0x79) {
+            sp->unk2 = str[2];
+            str += 3;
+            if (*str == 0x2E)
+                str++;
+        }
+        else if (str[1] == 0x3E) {
+            str = sub_800915C(&sp->unk0, str + 2);
+        }
+        else if (str[1] == 0x2E) {
+            sp->unk0 += str[2];
+            str += 3;
+        }
+        else if (str[1] == 0x6E) {
+            sp->unk0 = sp->unkC;
+            sp->unk2 += 11;
+            str += 2;
+        }
+        else if (str[1] == 0x3A) {
+            sp->unk0 = sp->unk4;
+            str += 2;
+        }
+        else if (str[1] == 0x3B) {
+            sp->unk0 = sp->unk4 + str[2];
+            str += 3;
+        }
+        else if (str[1] == 0x2B) {
+            str += 2;
+            sp->unk0 = (strArr[0].unk4 * 8) - sub_8008ED0(str);
+            sp->unk0 /= 2;
+        }
+        else if (str[1] == 0x43) {
+            sp->unk14 = sp->unk10;
+            sp->unk10 = InterpretColorChar(str[2]);
+            str += 3;
+        }
+        else if (str[1] == 0x5F) {
+            sp->unk14 = sp->unk10;
+            sp->unk10 = gUnknown_202749A[str[2]];
+            str += 3;
+            if (*str == 0x2E)
+                str++;
+        }
+        else if (str[1] == 0x52) {
+            sp->unk10 = sp->unk14;
+            str += 2;
+        }
+        else if (str[1] == 0x63) {
+            sp->unk18 = sp->unk10;
+            sp->unk10 = InterpretColorChar(str[2]);
+            str += 3;
+        }
+        else if (str[1] == 0x72) {
+            sp->unk10 = sp->unk18;
+            str += 2;
+        }
+        else if (str[1] == 0x53) {
+            gUnknown_20274A6[str[2] & 0x7F] = str[3] & 0x7F;
+            str += 4;
+        }
+        else if (str[1] == 0x57) {
+            str += 2;
+            sp->unk8 = ((strArr[0].unk0 * 8) + sp->unk0) - 2;
+            sp->unkA = ((strArr[0].unk2 * 8) + sp->unk2) + 3;
+            sp->unk20 = 1;
+            break;
+        }
+        else if (str[1] == 0x50) {
+            str += 2;
+            sp->unk2 = 9999;
+            sp->unk1C = 0;
+            sp->unk20 = 1;
+            break;
+        }
+        else if (str[1] == 0x70) {
+            str += 2;
+            sp->unk2 = 9999;
+            sp->unk1C = 1;
+            sp->unk20 = 1;
+            break;
+        }
+        else if (str[1] == 0x7E) {
```

Comment:
All these should be chars

Reply:
Looks like this function was missed when replacing character constants.

Reply:
I'll merge the PR and fix these in a follow-up PR.